Develop

Ushlerdop opened the pull request:
Создал две отдельных формы для создания и изменения задач. Добавил компоненты модального окна, в котором и изменяется задача. И кнопку для редактирования задачи. Модальное окно вызывается с помощью кнопки с карандашом у каждой задачи. Также покопался немного в стилях и добавил заготовку для ограничения текста в слишком больших задачах.
Изменил createdDate на editedDate. Теперь при создании и изменении будет отображаться 'Edited' и дата изменения. Также как и в Google Keep.

P.S. Извиняюсь за такой большой PR, по неопытности :( 
В следующий раз буду делать commit при меньших изменениях.

Ushlerdop committed:
Create-react-app done

Ushlerdop committed:
First working version

A 3 new components. 'Todo' contains the whole TODO app. 'TodoList' contains list of tasks. 'TodoForm' contains form for creating new tasks. The next step is create a new elements - TodoTask, input, button, etc.

Ushlerdop committed:
Created a TodoTask component

A new TodoTask component and module styles for it

Ushlerdop committed:
Created 3 new button components

Ushlerdop committed:
Added a responsive design layout

Ushlerdop committed:
Added a modal window for changing tasks

Создал две отдельных формы для создания и изменения задач. Добавил компоненты модального окна, в котором и изменяется задача. И кнопку для редактирования задачи. Модальное окно вызывается с помощью кнопки с карандашом у каждой задачи. Также покопался немного в стилях и добавил заготовку для ограничения текста в слишком больших задачах.

Ushlerdop committed:
Changes in TaskList styles and UpdateForm

Теперь карточки задач выстраиваются по 2 штуки в ряд на ПК. И по одной в ряд на мобильных устройствах.
Также добавил два пропса (isDone, isImportant) в UpdateForm для сохранения этих значений при изменении задачи

Ushlerdop committed:
Added TodoMods component

Добавил компонент TodoMods, в котором находятся кнопки для переключения показываемых задач. Логики ещё нет, только вёрстка

Ushlerdop committed:
Added a task filter

Теперь фильтр задач работает. Есть 4 выбора: все задачи, активные, важные и выполненные

Ushlerdop committed:
Changed Modal component to HOC

Сделал из компонента Modal HOC, чтобы вызывать в модальных окнах другие компоненты.

Ushlerdop committed:
Added a modal window for every task by click on them

Теперь при клике на каждую задачу открывается модальное окно с задачей без обрезанного текста в заголовке и описании.

Ushlerdop committed:
Some changes in styles

Изменил стили компонент с задачами и формы изменения задач

Ushlerdop committed:
Style changes

Теперь textarea в формах подстраивается под размер текста внутри. Также поменял стили полосы прокрутки у форм и у всего приложения для пк версии

Ushlerdop committed:
Added validation for forms

Добавил валидацию по количеству символов и пустоте инпутов для форм

Ushlerdop committed:
Added a check for the same title in other tasks

теперь нельзя изменить title задачи или создать новую задачу, если уже есть одна с таким же title.
При этом учитывается id, для избежения конфликта со своим же неизменённым тайтлом у задачи

Ushlerdop committed:
Added a little scale to Task on hover

Теперь карточки заданий немножко увеличиваются при наведении

Ushlerdop committed:
Change references to Ref in event handlers in Forms

Поменял рефы на e.target в обработчиках форм

Ushlerdop committed:
Added props with tasks for TodoUpdateForm in TodoTaskModal

Добавил пропс с заданиями в модальное окно с задачами. До этого при редактировании задачи вызывало ошибку undefined при валидации

Ushlerdop committed:
Added classes for active TodoMods buttons

Теперь выбранный фильтр по задачам выделяется цветом фона

Ushlerdop committed:
Changed colors of app

Изменил цвета приложения на более контрастные

Ushlerdop committed:
Added classnames library and used

"Победил" длинные условия для добавления классов библиотекой classnames. Теперь все условия хранятся в переменных, которые я передаю в className каждого элемента, где нужны условные классы. Выполненные задачи выделяются зелёным фоном, даже если они помечены как важные.
Изменил валидацию форм, теперь нужно ввести и тайтл, и описание

Ushlerdop committed:
Update README.MD

Добавил описание проекта

BubbaCat commented on src/components/Todo/Todo.jsx:
```diff
@@ -0,0 +1,115 @@
+  constructor(props) {
+    super(props);
+    this.state = {
+      tasks: [
```
Если решил замокать данные, то лучше вынеси их в отдельный файл

BubbaCat commented on src/components/Todo/Todo.jsx:
```diff
@@ -0,0 +1,115 @@
+
+  updateTask(id, updatedTask) {
+    this.setState({
+      tasks: this.state.tasks.map(task => {
```
const tasks = this.state.tasks.map(task => { .....})
this.SetState({tasks});

BubbaCat commented on src/components/Todo/Todo.jsx:
```diff
@@ -0,0 +1,115 @@
+
+  isDoneToggle(id) {
+    this.setState({
+      tasks: this.state.tasks.map(task => {
```
см. выше

BubbaCat commented on src/components/Todo/Todo.jsx:
```diff
@@ -0,0 +1,115 @@
+
+  isImportantToggle(id) {
+    this.setState({
+      tasks: this.state.tasks.map(task => {
```
см. выше

BubbaCat commented on src/components/Todo/TodoForms/TodoAddForm/TodoAddForm.jsx:
```diff
@@ -0,0 +1,116 @@
+
+    const maxTextInputLength = 500;
+    const maxTextAreaLength = 3000;
+    if ( e.target.title.value && e.target.description.value ) {
```
Лучше не делать вложенность if'ов
    if ( !(e.target.title.value && e.target.description.value) ) { 
          alert('You have to text something in Title and Description');
          return;
    } ну и потом уже следующий if написать

BubbaCat commented on src/components/Todo/TodoForms/TodoAddForm/TodoAddForm.module.scss:
```diff
@@ -0,0 +1,104 @@
+.todoForm {
+  border-radius: 10px;
+  &Container {
+    border-radius: inherit;
```
Лучше писать css стили в от в таком порядке 
1.Сначала положение элемента относительно других
2. Затем размеры и отступы
3.Рамка border, она частично относится к размерам
4.Общее оформление содержимого
5.Цветовое и стилевое оформление

так легче читать и расширять, но это просто заметка на будущее, можно поставить npm пакет, которые это будет делать за нас в одном команду

BubbaCat commented on src/components/Modal/Modal.jsx:
```diff
@@ -0,0 +1,26 @@
+function Modal(Component) {
+  return class extends React.Component {
+    componentDidMount() {
+    }
```
?

BubbaCat commented on src/components/Modal/Modal.jsx:
```diff
@@ -0,0 +1,26 @@
+    render() {
+      return (
+        <div
+          className={this.props.active ? styles.modal + ' ' + styles.active : styles.modal}
```
\` ${styles.modal} ${styles.active} `

BubbaCat replied:
У тебя же classNames подключен, используй его для таких условий

BubbaCat commented on src/components/Modal/Modal.jsx:
```diff
@@ -0,0 +1,26 @@
+          onClick={() => this.props.setModalActive(false)}
+        >
+          <div
+            className={this.props.active ? styles.modalContent + ' ' + styles.active : styles.modalContent}
```
см выше

BubbaCat commented on src/components/Todo/TodoList/TodoList.jsx:
```diff
@@ -0,0 +1,70 @@
+        />
+        <ul className={styles.todoList}>
+          {this.state.currentTasks.map(task => (
+            <TodoTask
```
т.к у тебя поля совпадают поля пропсов совпадают с названиями полей класса то можно
<TodoTask
key={task.id}
{...task}
/>
js позволяет не писать одинаковые поля для объектов : 
{
tast: task,
id: id,
} 
тоже самое что и(при условия, что те поля не undefined)
{
task,
id
}

BubbaCat commented on src/components/Todo/TodoTask/TodoTask.jsx:
```diff
@@ -0,0 +1,111 @@
+        <TodoUpdateForm
+          active={this.state.isEditModalActive}
+          setModalActive={this.setEditModalActive}
+          updateTask={this.props.updateTask}
```
{...this.props}

Ushlerdop replied:
В пропсах находятся методы, которые не используются в этом компоненте. Или ради читаемости кода можно передать немного лишнего в компонент?

BubbaCat commented on src/components/Todo/TodoTask/TodoTask.jsx:
```diff
@@ -0,0 +1,111 @@
+        <TodoTaskModal
+          active={this.state.isInfoModalActive}
+          setModalActive={this.setInfoModalActive}
+          {...this.props}
```
можешь ведь, когда захочешь :)

BubbaCat commented on src/components/Todo/TodoTask/TodoTaskModal.jsx:
```diff
@@ -0,0 +1,96 @@
+        <TodoUpdateForm
+          active={this.state.isEditModalActive}
+          setModalActive={this.setEditModalActive}
+          updateTask={this.props.updateTask}
```
см выше

BubbaCat reviewed (commented):
В целом все хорошо, молодец :)

BubbaCat commented on src/components/Todo/TodoMods/TodoMods.jsx:
```diff
@@ -0,0 +1,88 @@
+      todoModsButtonActive: this.state.activeTaskFilterId === 3,
+    });
+
+    let DoneTasksClassName = cx({
```
const

BubbaCat commented on src/components/Todo/TodoMods/TodoMods.jsx:
```diff
@@ -0,0 +1,88 @@
+      todoModsButtonActive: this.state.activeTaskFilterId === 2,
+    });
+
+    let ImportantTasksClassName = cx({
```
const и в езде используй const

BubbaCat commented on src/components/Todo/TodoTask/TodoTask.jsx:
```diff
@@ -0,0 +1,111 @@
+import styles from './TodoTask.module.scss';
+import classNames from 'classnames/bind';
+
+let cx = classNames.bind(styles);
```
const

BubbaCat commented on src/components/Todo/TodoTask/TodoTaskModal.jsx:
```diff
@@ -0,0 +1,96 @@
+      taskDone: this.props.isDone,
+    });
+
+    let titleTaskClassName = cx({
```
let на const во всех местах

BubbaCat commented on src/components/Todo/Todo.jsx:
```diff
@@ -0,0 +1,115 @@
+  }
+
+  isDoneToggle(id) {
+    this.setState({
```
Ну на самом деле лучше в setState функцию (prevState)=> 
Чтобы не было потом, что у нас стейт ведет себя непредсказуемо.

BubbaCat commented:
Также доп замечания  от Михаила  написаны в телеграм

Ushlerdop committed:
Changed the code to a more readable one, part 1

Изменил код на более читаемый, ч.1
Добавил файл tasks с массивом, в котором хранятся задачи

Ushlerdop committed:
changed the code based on comments from code review

Исправил некоторые замечания из код ревью. Осталось объеденить стили для Add и Update форм.

Ushlerdop committed:
Now Add & Update forms use the same style component

Объединил файл стилей в один для Add и Update форм.

Ushlerdop committed:
Some changes in form's style component

Вернул автоматическую подстройку высоты textarea у updateForm, т.к. случайно удалил
